src/app/files/fileManager.js

```diff
@@ -42,6 +42,212 @@ class FileManager extends Plugin {
     this.init()
   }
 
+    /**
+   * Emit error if path doesn't exist
+   * @param {string} path path of the file/directory
+   * @param {string} message message to display if path doesn't exist.
+   */
+  _handleExists (path, message) {
+    if (!this.exists(path)) {
+      this._handleError({ code: 'ENOENT', message })
+    }
+  }
+
+  /**
+   * Emit error if path is not a file
+   * @param {string} path path of the file/directory
+   * @param {string} message message to display if path is not a file.
+   */
+  _handleIsFile (path, message) {
+    if (!this.isFile(path)) {
+      this._handleError({ code: 'EISDIR', message })
+    }
+  }
+
+  /**
+   * Emit error if path is not a directory
+   * @param {string} path path of the file/directory
+   * @param {string} message message to display if path is not a directory.
+   */
+  _handleIsDir (path, message) {
+    if (this.isFile(path)) {
+      throw new Error({ code: 'ENOTDIR', message })
+    }
+  }
+
+  /**
+   * Emits error based on error code
+   * @param {object} error error { code, message }
+   */
+  _handleError (error) {
+    const message = error.message ? `: ${error.message}` : ''
+
+    if (error.code === 'ENOENT') {
+      throw new Error('No such file or directory' + message)
+    }
+
+    if (error.code === 'EISDIR') {
+      throw new Error('Path is a directory' + message)
+    }
+
+    if (error.code === 'ENOTDIR') {
+      throw new Error('Path is not on a directory' + message)
+    }
+
+    if (error.code === 'EEXIST') {
+      throw new Error('File already exists' + message)
+    }
+
+    if (error.code === 'EPERM' || error.code === 'EACCESS') {
+      throw new Error('Permission denied' + message)
+    }
+
+    return error
+  }
+
+  /** The current opened file */
+  file () {
+    const file = this.currentFile()
+
+    if (!file) this._handleError({ code: 'ENOENT', message: 'No file selected' })
+    return file
+  }
+
+  /**
+   * Verify if the path exists (directory or file)
+   * @param {string} path path of the directory or file
+   * @returns {boolean} true if the path exists
+   */
+  exists (path) {
+    const provider = this.fileProviderOf(path)
+    return provider ? true : false // eslint-disable-line
+  }
+
+  /**
+   * Verify if the path provided is a file
+   * @param {string} path path of the directory or file
+   * @returns {boolean} true if path is a file.
+   */
+  isFile (path) {
+    const extension = path.split('/').pop()
+
+    return extension && extension.indexOf('.') > -1
```

Comment:
you can have browser/myfolder.is.cool folder

Reply:
i think you should ask provider for this

Reply:
Okay